### PR TITLE
[Silabs] Fixed LCD animations for window-app

### PR DESCRIPTION
**Pull request by arun-silabs**
Issue : LCD animations are not working for window-app.

Fix : The variable 'mState.isWiFiProvisioned' is not being modified after commissioning, so the LCD 'paint' API was not being called. Replaced the variable with respective API call to get the commissioning status. 

After factory reset, QR code is also being displayed on LCD.

Notes : For LCD to work on thread applications, the variable 'mState.isThreadProvisioned' must be set to 1 after commissioning. We were unable to verify this as we do not have the thread setup.

### CODE REVIEW

File: examples/window-app/silabs/src/WindowManager.cpp [hunk 1]
```diff
@@ -738,13 +739,22 @@ void WindowManager::UpdateLCD()
...
         }
-#endif
     }
+#ifdef QR_CODE_ENABLED
+    else
+    {
+        chip::MutableCharSpan qrCode(mQRCodeBuffer);
+        if (GetQRCode(qrCode, chip::RendezvousInformationFlags(chip::RendezvousInformationFlag::kBLE)) == CHIP_NO_ERROR)
+        {
+            slLCD.SetQRCode((uint8_t *) qrCode.data(), qrCode.size());
```
Comment (bzbarsky-apple) [inline examples/window-app/silabs/src/WindowManager.cpp:753]: Please use Uint8::from_char or Uint8::from_const_char as appropriate here.

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/window-app/silabs/src/WindowManager.cpp (hunk 1 above)
Comment (arun-silabs) [reply examples/window-app/silabs/src/WindowManager.cpp:753]: Sure, will update.

File: examples/window-app/silabs/include/WindowManager.h [hunk 2]
```diff
@@ -160,7 +160,8 @@ class WindowManager
 
     LEDWidget mStatusLED;
     LEDWidget mActionLED;
-
+    // Get QR Code and emulate its content using NFC tag
```
Comment (jmartinez-silabs) [inline examples/window-app/silabs/include/WindowManager.h:163]: This comment is pretty odd since we don't have a NFC

File: examples/window-app/silabs/include/WindowManager.h [hunk 3]
```diff
@@ -160,7 +160,8 @@ class WindowManager
...
+    char mQRCodeBuffer[chip::QRCodeBasicSetupPayloadGenerator::kMaxQRCodeBase38RepresentationLength + 1];
```
Comment (jmartinez-silabs) [inline examples/window-app/silabs/include/WindowManager.h:164]: This doesn't seem to match how our other applications work.

File: examples/window-app/silabs/src/WindowManager.cpp [hunk 4]
```diff
@@ -665,7 +665,7 @@ void WindowManager::UpdateLEDs()
 #if CHIP_ENABLE_OPENTHREAD
             if (mState.isThreadProvisioned && mState.isThreadEnabled)
 #else
-            if (mState.isWiFiProvisioned && mState.isWiFiEnabled)
+            if (ConnectivityMgr().IsWiFiStationProvisioned() && ConnectivityMgr().IsWiFiStationEnabled())
```
Comment (jmartinez-silabs) [inline examples/window-app/silabs/src/WindowManager.cpp:668]: Why not keep variables and set them correctly. This should be done for all apps in base applications if I am not mistaken

File: examples/window-app/silabs/src/WindowManager.cpp [hunk 5]
```diff
@@ -725,7 +725,7 @@ void WindowManager::UpdateLCD()
 #if CHIP_ENABLE_OPENTHREAD
     if (mState.isThreadProvisioned)
 #else
-    if (mState.isWiFiProvisioned)
+    if (ConnectivityMgr().IsWiFiStationProvisioned())
```
Comment (jmartinez-silabs) [inline examples/window-app/silabs/src/WindowManager.cpp:728]: Same comment here

File: examples/window-app/silabs/src/WindowManager.cpp [hunk 6]
```diff
@@ -739,13 +739,22 @@ void WindowManager::UpdateLCD()
         Attributes::CurrentPositionTilt::Get(cover.mEndpoint, tilt);
         chip::DeviceLayer::PlatformMgr().UnlockChipStack();
 
-#ifdef DISPLAY_ENABLED
```
Comment (jmartinez-silabs) [inline examples/window-app/silabs/src/WindowManager.cpp:742]: Why do we remove the ifdef? It seems ok to not call the LCD paint if the display is not enabled

Comment (arun-silabs) [reply examples/window-app/silabs/src/WindowManager.cpp:742]: This check is duplicate one as the same ifdef is present just above in the code. (Line #724)

File: examples/window-app/silabs/src/WindowManager.cpp (hunk 4 above)
Comment (arun-silabs) [reply examples/window-app/silabs/src/WindowManager.cpp:668]: Earlier, in common WindowApp.cpp, these variables are being set in 'Run()' API. This API is missing in WindowManager.cpp after refactoring. Since there are only 1-2 instances of these variables, we have removed them and directly added APIs.

File: examples/window-app/silabs/src/WindowManager.cpp [hunk 7]
```diff
@@ -739,13 +739,22 @@ void WindowManager::UpdateLCD()
...
         if (!tilt.IsNull() && !lift.IsNull())
         {
             LcdPainter::Paint(slLCD, type, lift.Value(), tilt.Value(), mIcon);
         }
-#endif
     }
+#ifdef QR_CODE_ENABLED
+    else
+    {
+        chip::MutableCharSpan qrCode(mQRCodeBuffer);
```
Comment (jepenven-silabs) [inline examples/window-app/silabs/src/WindowManager.cpp:750]: This seems like duplication. Since the Window-app was migrated to use the BaseApplication all of the content present in this PR should theoritically be already present in the BaseApplication..... Last time I've check everything was working fine with the Window-app Thread.... Why is this needed at all ?

**Review by jepenven-silabs — CHANGES_REQUESTED** [review]
Requesting changes to prevent accidental merge..

File: examples/window-app/silabs/src/WindowManager.cpp (hunk 6 above)
Comment (jmartinez-silabs) [reply examples/window-app/silabs/src/WindowManager.cpp:742]: ah thank you!

File: examples/window-app/silabs/src/WindowManager.cpp (hunk 7 above)
Comment (jmartinez-silabs) [reply examples/window-app/silabs/src/WindowManager.cpp:750]: Jf is right here. the base application is the one dealing with the QR code. See Function
`BaseApplication::OutputQrCode(bool refreshLCD)` and its use.

File: examples/window-app/silabs/src/WindowManager.cpp (hunk 4 above)
Comment (jmartinez-silabs) [reply examples/window-app/silabs/src/WindowManager.cpp:668]: If not used at all, I would expect the variables to be deleted all together our `WindowManager.h`
Also calling the following apis `ConnectivityMgr().IsWiFiStationProvisioned() and ConnectivityMgr().IsWiFiStationEnabled()` might work for wifi, But it I problematic to call it frequently for thread ( slows the system a lot due to nvm3 reads for the dataset)

BaseApplication.cpp already deals with checking the provision state in a safe way. I would leverage that instead.

Lastly, this whole UpdateLEDs() function in the window-app seems to counter what is done in the `baseApplication` `LightTimerEventHandler`. That is definitely and issue.

The rework of this example might have missed some key elements. We have to fix this correctly so it behaves like all our examples

File: examples/window-app/silabs/src/WindowManager.cpp (hunk 7 above)
Comment (arun-silabs) [reply examples/window-app/silabs/src/WindowManager.cpp:750]: Noticed that this part of code is not required for displaying QR. Will remove this.

File: examples/window-app/silabs/include/WindowManager.h (hunk 2 above)
Comment (arun-silabs) [reply examples/window-app/silabs/include/WindowManager.h]: Removed this as it is no longer needed.

File: examples/window-app/silabs/include/WindowManager.h (hunk 3 above)
Comment (arun-silabs) [reply examples/window-app/silabs/include/WindowManager.h]: Removed this as it is no longer needed.

File: examples/window-app/silabs/src/WindowManager.cpp (hunk 5 above)
Comment (arun-silabs) [reply examples/window-app/silabs/src/WindowManager.cpp]: Replaced this to get provision status from BaseApplication.cpp

File: examples/window-app/silabs/src/WindowManager.cpp (hunk 4 above)
Comment (arun-silabs) [reply examples/window-app/silabs/src/WindowManager.cpp]: Thanks for the comments.

**LightTimerEventHandler** function in BaseApplication.cpp can be used to handle Status LED (LED0), UpdateLEDs() function will be needed for window-app to deal with ActionLED (LED 1). Removed the code which handles LED0 in UpdateLEDS() function .

**Review by jmartinez-silabs — APPROVED** [review]

**Review by jepenven-silabs — APPROVED** [review]
